kroxylicious-filters/kroxylicious-record-validation/src/main/java/io/kroxylicious/proxy/filter/validation/ProduceValidationFilterFactory.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.validation;
+
+import io.kroxylicious.proxy.filter.validation.config.ValidationConfig;
+import io.kroxylicious.proxy.plugin.Plugin;
+
+@Plugin(configType = ValidationConfig.class)
+@Deprecated(since = "0.7.0", forRemoval = true)
+public class ProduceValidationFilterFactory extends RecordValidation {
```

Comment:
suggest add a warning to the default constructor:

```
public ProduceValidationFilterFactory() {
        super();
        LOGGER.warn("ProduceValidationFilterFactory is deprecated, use 'RecordValidation' in your Filter configuration instead");
    }
```

Reply:
The plugin mechanism already logs that for us:

```
2024-08-16 09:19:21 WARN  io.kroxylicious.proxy.config.ServiceBasedPluginFactoryRegistry:101 - io.kroxylicious.proxy.filter.FilterFactory plugin with id ProduceValidationFilterFactory is deprecated
```

I've added a `@deprecated tag to the javadoc to allow users to find the replacement.